android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt

```diff
@@ -95,6 +96,10 @@ class VideoPlaybackService : MediaSessionService() {
 
     override fun onDestroy() {
         cleanup()
+        val notificationManager: NotificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
```

Comment:
I think that would cause a problem because the foreground service/notification could be still alive when cleanup() is ran. I can test this though and report back.

Reply:
After double check, this code doesn't seem necessary:
cleanup() function calls hideAllNotifications() which is doing exactly the same code  + notificationManager.cancelAll()

I think it can just be removed in fact

Reply:
In fact I am disturbed by the logic.
Notification channel is created in onUpdateNotification and destroyed in onDestroy

For exemple shouldn't you add this code also in : (which calls cleanup)
unregisterPlayer(player: ExoPlayer) { 

in fact (maybe), you need to replace cleanup() by hideAllNotifications() in onTaskRemoved and unregisterPlayer() ?

Notice that if you unregisterPlayer and registerPlayer just after the notification channel is not re created...

onDestroy seems ok for me to cleanup.

Reply:
@paulrinaldi gentle reminder on this point, It would be great to merge this PR for next release !